source/agora/consensus/EnrollmentManager.d

```diff
@@ -544,6 +559,26 @@ public class EnrollmentManager
         return enrolls;
     }
 
+    /***************************************************************************
+
+        Get a pre-image for revelation
+
+        Params:
+            preimage = will contain the PreimageInfo if exists
+
+        Returns:
+            true if the pre-image exists
+
+    ***************************************************************************/
+
+    public bool getNextPreimage (out PreimageInfo preimage) @safe
+    {
+        auto height = this.next_reveal_height + PreimageRevealPeriod * 2;
```

Comment:
Why the `* 2` part? So will this actually reveal preimages for the next 12 blocks (2 hours), rather than 6 blocks (1 hour)?

Reply:
That was discussed during the whiteboarding session. Currently it's a bit convoluted, as you mentioned, we should use a sliding windows instead.

Reply:
Thanks, I missed the first part of the whiteboarding session when this happened.